src/libstd/sys_common/wtf8.rs

```diff
@@ -615,8 +615,8 @@ impl Wtf8 {
         if len < 3 {
             return None;
         }
-        match &self.bytes[..3] {
-            &[0xED, b2 @ 0xB0..=0xBF, b3] => Some(decode_surrogate(b2, b3)),
+        match self.bytes[..3] {
```

Comment:
The subslice pattern could be used here.

Reply:
I'm not too familiar with subslice patterns yet so I'll pass.

Reply:
Resolving it in #70046